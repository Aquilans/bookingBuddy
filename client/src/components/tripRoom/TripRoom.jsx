import React from 'react';
import {tripData, hotelRecomendations} from './tripRoomDummyData';
import TripRecomendationsCards from './tripRecomendationsCards.jsx';
import GroupPreferencesBar from './groupPreferencesBar';

/////////////////////////
// Trip Room Components
/////////////////////////

class TripRoomComponents extends React.Component {

  constructor (props) {
    super(props);
    this.setLocation = this.setLocation.bind(this);
    this.addBookmark = this.addBookmark.bind(this);
    this.state = {
      priceRange: props.tripData.priceRange,
      dateRange: props.tripData.dateRange,
      locations: props.tripData.locations,
      selectedLocation: '',
      bookmarkedTrips: props.tripData.bookmarkedTrips.slice()
    };
  }

  setLocation(selection) {
    this.setState({
      selectedLocation: selection
    });
  }

  addBookmark(newBookmark) {
    newBookmark['buddyVotes'] = this.props.tripData.buddyList.map((buddyName) => ({
      buddyName: buddyName,
      buddyVote: 0
    }));
    this.setState({
      bookmarkedTrips: this.state.bookmarkedTrips.concat(newBookmark)
    });
  }

  render() {

    return (
      <div className="container">
        <h1 className="orange-text darken-2">{this.props.tripData.tripName}</h1>
        <GroupPreferencesBar
          priceRange={this.state.priceRange}
          dateRange={this.state.dateRange}
          locations={this.state.locations}
          setLocation={this.setLocation}
        />
        <TripRecomendationsCards
          hotelRecomendations={this.props.hotelRecomendations}
          addBookmark={this.addBookmark}
        />
      </div>

      //
      // <TripBookmarksList
      //   bookmarkedTrips=""
      // />
    );
  }

};

class TripRoom extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  componentDidMount() {
    var obj = {};
    $.ajax({
      type: 'POST',
      url: '/getTotal',
      dataType: 'json',
      data: {'id': 1},
      success: function(comments) {
        obj['sum'] = comments.sum;
        console.log(comments, 'RESPONSE!');
        this.setState({budgetSum: comments.sum});
      }.bind(this)
    });
    $.ajax({
      type: 'POST',
      url: '/commonTrip',
      dataType: 'json',
      data: {'id': 1},
      success: function(comments) {
        obj.location = comments.commonTrips;
        console.log(comments, 'COMMON');
        this.setState({commonLocation: comments.commonTrips});
      }.bind(this)
    });
    $.ajax({
      type: 'POST',
      url: '/commonDate',
      dataType: 'json',
      data: {'id': 1},
      success: function(comments) {
        obj.dates = [comments.beginning, comments.ending, comments.duration];
        console.log(comments, 'COMMON DATES');
        this.setState({commonDateB: comments.beginning, commonDateE: comments.ending});
      }.bind(this)
    });

    setTimeout(function() {
      $.ajax({
        type: 'POST',
        url: '/hotwire',
        dataType: 'json',
        data: obj,
        success: function(data) {
          console.log(data);
        }.bind(this)
      });
    }, 1000);
  }

  render() {
    return (
      <div>
        <TripRoomComponents
          tripData={tripData}
          hotelRecomendations={hotelRecomendations}
        />
      </div>
    );
  }

  renderComment({body, author}) {
    return (
      <div>
        <li></li>;
      </div>
    );
  }
}
export default TripRoom;

