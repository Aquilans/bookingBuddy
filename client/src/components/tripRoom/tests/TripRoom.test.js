import React from 'react';
import renderer from 'react-test-renderer';

import TripRoom from '../TripRoom.jsx';

it('renders correctly', () => {
  const tripRoom = renderer.create(
    <TripRoom />
  ).toJSON();
  expect(tripRoom).toMatchSnapshot();
});