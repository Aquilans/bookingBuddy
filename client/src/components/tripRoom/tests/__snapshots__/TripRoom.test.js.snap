// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <h1
      className="orange-text darken-2"
    >
      Hiking Trip
    </h1>
    <div>
      <div
        className="row"
      >
        <div
          className="col s3"
        >
          <span>
            <i
              className="material-icons green-text"
            >
              credit_card
            </i>
            $1,500-$2,500
          </span>
          <br />
        </div>
        <div
          className="col s3"
        >
          <span>
            <i
              className="material-icons green-text"
            >
              today
            </i>
            12/10 - 12/21
          </span>
          <br />
          <br />
        </div>
      </div>
      <span>
        <strong>
          Based on your group's preferences, we found all of these trips for you!
        </strong>
        <select
          onChange={[Function]}
          value=""
        >
          <option
            value="Hawaii"
          >
             
            Hawaii
             
          </option>
          <option
            value="Florida"
          >
             
            Florida
             
          </option>
          <option
            value="Bahamas"
          >
             
            Bahamas
             
          </option>
        </select>
      </span>
    </div>
    <div
      className="row"
    >
      <div
        className="col s12 m8 l6"
      >
        <div
          className="card"
        >
          <div>
            <div
              className="card-image"
              style={
                Object {
                  "backgroundImage": "url(https://s3-media3.fl.yelpcdn.com/bphoto/8qcpzDf8VSeYxPtHG4Lu5g/o.jpg)",
                  "backgroundPosition": "center",
                  "backgroundSize": "cover",
                  "height": "400px",
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "width": "450px",
                }
              }
            >
              <span
                className="card-title"
              >
                Sagamore Pendry Baltimore
              </span>
            </div>
            <div
              className="card-content"
            >
              <p>
                <i
                  className="material-icons orange-text"
                >
                  star star star star star 
                </i>
                
              </p>
              <p>
                Where all the ballers and shot callers come to relax and recharge
              </p>
              <p>
                $
                450.91
                 total per person
              </p>
            </div>
          </div>
          <div
            className="card-action"
          >
            <a
              onClick={[Function]}
            >
              Next
            </a>
          </div>
        </div>
      </div>
      <div>
        <form
          onSubmit={[Function]}
        >
          <textarea
            className="materialize-textarea"
            defaultValue=""
            onChange={[Function]}
            placeholder="Want to stay here? Write a quick note to your buddies about why you like this one, then bookmark it to highlight this option for your friends to see!"
          />
           
          <br />
          <button
            className="btn-large waves-effect waves-light orange"
            onClick={[Function]}
            type="submit"
          >
            Add Bookmark!
          </button>
        </form>
      </div>
    </div>
    <div />
  </div>
</div>
`;
